### PR TITLE
Fix JointStatePublisher topic name for nested models

### PR DESCRIPTION
**Pull request by iche033**
Signed-off-by: Ian Chen <ichen@osrfoundation.org>

# 🦟 Bug fix

## Summary

The joint-state-publisher system currently assumes it's always attached to the top level model and hence generates an incorrect topic name for nested models. This PR updates it to generate the correct topic name.

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by chapulina — APPROVED** [review]

File: test/integration/joint_state_publisher_system.cc [hunk 1]
```diff
@@ -144,3 +144,58 @@ TEST_F(JointStatePublisherTest,
...
 }
+
+/////////////////////////////////////////////////
+TEST_F(JointStatePublisherTest,
+       IGN_UTILS_TEST_DISABLED_ON_WIN32(NestedJointPublisher))
+{
+  // Start server
+  ServerConfig serverConfig;
+  serverConfig.SetSdfFile(std::string(PROJECT_SOURCE_PATH) +
+      "/test/worlds/diff_drive_nested.sdf");
```
Comment (ahcorde) [inline test/integration/joint_state_publisher_system.cc:155]: `ignition::common::joinPaths`

File: test/integration/joint_state_publisher_system.cc [hunk 2]
```diff
@@ -144,3 +144,58 @@ TEST_F(JointStatePublisherTest,
...
+  EXPECT_FALSE(*server.Running(0));
+
+  server.SetUpdatePeriod(0ns);
+
+  int count = 0;
+  // Check that all of joints are published.
+  std::function<void(const msgs::Model &)> jointStateCb =
+    [&](const msgs::Model &_msg)
+    {
+      std::cerr << "asdfasdf" << std::endl;
```
Comment (ahcorde) [inline test/integration/joint_state_publisher_system.cc:168]: remove

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: test/integration/joint_state_publisher_system.cc (hunk 2 above)
Comment (iche033) [reply test/integration/joint_state_publisher_system.cc]: oops, removed. [8f540bf](https://github.com/ignitionrobotics/ign-gazebo/pull/1405/commits/8f540bfdd73a472665b3e2c08a96ece1a43880c0)

File: test/integration/joint_state_publisher_system.cc (hunk 1 above)
Comment (iche033) [reply test/integration/joint_state_publisher_system.cc]: done. [8f540bf](https://github.com/ignitionrobotics/ign-gazebo/pull/1405/commits/8f540bfdd73a472665b3e2c08a96ece1a43880c0)

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-04-13-fortress-edifice/1367/1

**Pull request merged**